sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -495,8 +480,11 @@ object ScalaJSPluginInternal {
 
   val scalaJSTestFrameworkSettings = Seq(
       loadedTestFrameworks := {
-        // use assert to prevent warning about pure expr in stat pos
-        assert(scalaJSEnsureUnforked.value)
+        if (fork.value) {
+          throw new MessageOnlyException(
+              "`test` tasks in a Scala.js project require " +
+              "`fork in Test := false`.")
+        }
```

Comment:
Oh, so things actually *do* go wrong if we try and run the test stuff in a forked JVM? If so, I'm fine with having it here.